nixos/modules/services/networking/firewalld/default.nix

```
@@ -0,0 +1,62 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.firewalld;
+  paths = pkgs.buildEnv {
+    name = "firewalld-paths";
+    paths = cfg.packages;
+    pathsToLink = [ "/lib/firewalld" ];
+  };
+in
+{
+  imports = [
+    ./service.nix
+    ./settings.nix
+    ./zone.nix
+  ];
+
+  options.services.firewalld = {
+    enable = lib.mkEnableOption "FirewallD";
+    package = lib.mkPackageOption pkgs "firewalld" { };
+    packages = lib.mkOption {
+      type = lib.types.listOf lib.types.package;
+      description = "Packages providing firewalld zones and other files.";
+    };
+    extraArgs = lib.mkOption {
+      type = lib.types.listOf lib.types.str;
+      default = [ ];
+      example = [ "--debug" ];
+      description = "Extra arguments to pass to FirewallD.";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+    services.dbus.packages = [ cfg.package ];
+    services.firewalld.packages = [ cfg.package ];
+
+    services.logrotate.settings."/var/log/firewalld" = {
+      copytruncate = true;
+      minsize = "1M";
+    };
+
+    environment.etc."sysconfig/firewalld".text = ''
+      FIREWALLD_ARGS=${lib.escapeShellArgs cfg.extraArgs}
+    '';
+
+    systemd.packages = [ cfg.package ];
+    systemd.services.firewalld = {
+      aliases = [ "dbus-org.fedoraproject.FirewallD1.service" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig.ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
```

Nit: Can we prefer `getExe` and `getExe' "foo"` to `/bin/foo`?

Also a nit: but all my time running into shellcheck lint errors makes seeing `$foo` usage in shell args very painful; variable usage should almost always be quoted as `"$foo"` 😅 

```suggestion
      serviceConfig.ExecReload = ''
        ${lib.getExe' "kill" pkgs.coreutils} -HUP "$MAINPID"
      '';
```